lib/dao/payment_dao.dart

```diff
@@ -103,4 +105,194 @@ class PaymentDao {
     );
     return result;
   }
+
+  Future<int> findPaymentCategoryByTitle(String title) async {
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'title = ? AND autoCategorizationEnabled = ?',
+      whereArgs: [title, true], // Assuming you want to filter for true
+    );
+    if (maps.isNotEmpty) {
+
+      // Returns the Index of Applied Category
+      return (maps.first)['category']-1;
+    }
+    return 9;
+  }
+
+  Future<List<Map<String,dynamic>>?> findPaymentsWithMissingCategory() async {
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'category = ?',
+      whereArgs: [9], // Assuming you want to filter for true
+    );
+    if (maps.isNotEmpty) {
+      return maps;
+    }
+    return null;
+  }
+
+  Future<Map<String,dynamic>?> searchForSameTitle(String title,int id) async{
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'title = ? AND id != ?',
+      whereArgs: [title,id], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps.first;
+    }
+
+    return null;
+  }
+
+
+  Future<List<Map<String,dynamic>>?> getAllMiscellanous() async{
+
+    List<Category> allCategories = await CategoryDao().find();
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'category = ?',
+      whereArgs: [10], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps;
+    }
+
+    return null;
+  }
+
+
+  Future<int> updateMiscellanousCategories() async{
+
+    int count = 0;
+    CategoryDao categoryDao = CategoryDao();
+    AccountDao accountDao = AccountDao();
+    List<Map<String,dynamic>> allMisPayments = await getAllMiscellanous() ?? [];
+    if(allMisPayments.isNotEmpty){
+
+      for(Map<String,dynamic> misPayment in allMisPayments){
+        Map<String,dynamic> gotPayment = await searchForSameTitle(misPayment['title'], misPayment['id']) ?? {};
+        if(gotPayment.isNotEmpty){
+          Category? cat = await categoryDao.findCategoryById(gotPayment['category']);
+          Account? acc = await accountDao.findCategoryById(misPayment['account']);
+          final updatedMap = {"id": misPayment['id'], "title": misPayment['title'], "description": misPayment['description'], "account": acc!.toJson(), "category": cat!.toJson(), "amount": misPayment['account'].toDouble(), "type": misPayment['type'], "datetime": misPayment['datetime'], "autoCategorizationEnabled": misPayment['autoCategorizationEnabled'] == 1 ? true : false};
+          await upsert(Payment.fromJson(updatedMap),);
+          count++;
+        }
+      }
+
+    }
+    return count;
+  }
+
+  Future<int> categorizeUsingRules()async{
```

Comment:
Only if I use a title exactly like in titles hard coded array, categorization is working. And the amount is changing to 1.0.
![Screenshot_20240820_125159](https://github.com/user-attachments/assets/c524a7ce-705c-4eea-8f3c-7427cc738137)
![Screenshot_20240820_125149](https://github.com/user-attachments/assets/db706956-de5b-43a6-8027-ff1485b72b8f)